Add deprecation notice to otelmongo (#5598)
open-telemetry/opentelemetry-go-contrib: CHANGELOG.md

### Deprecated

- The `InterceptorFilter` type in `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` is deprecated. (#5196)
- The `go.opentelemetry.io/contrib/instrumentation/go.mongodb.org/mongo-driver/mongo/otelmongo` package is deprecated.

>   If you would like to become an owner of this module and prevent it from being frozen, see [#5551]. (#5598)

Reviewer: 1. I suggest using the term "Code Owner" like here: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/CONTRIBUTING.md#code-owners
2. I think "frozen" can be understood as "does not receive features but is still released".  Especially we already have such definition in OTel Spec: https://opentelemetry.io/docs/specs/otel/document-status/#feature-freeze. I would rather use "removed" (or "dropped", "abandoned").

```suggestion
  If you would like to become an Code Owner of this module and prevent it from being removed, see [#5551]. (#5598)
```

Author: ```suggestion
  If you would like to become a Code Owner of this module and prevent it from being removed, see [#5551]. (#5598)
```
